lib/pacemaker/pcmk_output_utils.c

```diff
@@ -43,3 +43,37 @@ pcmk__out_epilogue(pcmk__output_t *out, xmlNodePtr *xml, int retval) {
 
     pcmk__output_free(out);
 }
+
+/*!
+ * \internal
+ * \brief Create a new output object using the "log" format
+ *
+ * Create a new output object using the "log" format, and register the
+ * libpe_status and libpacemaker messages.
+ *
+ * \return Newly created output object, or NULL on error
+ */
+pcmk__output_t *
+pcmk__new_logger(void)
+{
+    int rc = pcmk_rc_ok;
+    pcmk__output_t *out = NULL;
+    const char* argv[] = { "", NULL };
+    pcmk__supported_format_t formats[] = {
+        PCMK__SUPPORTED_FORMAT_LOG,
+        { NULL, NULL, NULL }
+    };
+
+    pcmk__register_formats(NULL, formats);
+    rc = pcmk__output_new(&out, "log", NULL, (char**)argv);
+    if ((rc != pcmk_rc_ok) || (out == NULL)) {
+        crm_err("Can't log resource details due to internal error: %s\n",
```

Comment:
This was in the original code, but if you're moving it you might as well fix it at the same time - there's no need for a newline at the end of the log message.

Reply:
Good catch, added a commit on top